pandas/io/formats/style.py

```diff
@@ -1635,78 +1637,390 @@ def highlight_null(
         color: str = "yellow",
         axis: Optional[Axis] = 0,
+        props: Optional[str] = None,
     ) -> Styler:
         """
-        Highlight the minimum by shading the background.
+        Highlight the minimum with a style.
 
         Parameters
         ----------
         subset : IndexSlice, default None
             A valid slice for ``data`` to limit the style application to.
         color : str, default 'yellow'
+            Background color to use for highlighting.
         axis : {0 or 'index', 1 or 'columns', None}, default 0
             Apply to each column (``axis=0`` or ``'index'``), to each row
             (``axis=1`` or ``'columns'``), or to the entire DataFrame at once
             with ``axis=None``.
+        props : str, default None
+            CSS properties to use for highlighting. If ``props`` is given, ``color``
+            is not used.
+
+            .. versionadded:: 1.3.0
 
         Returns
         -------
         self : Styler
+
+        See Also
+        --------
+        Styler.highlight_null: Highlight missing values with a style.
+        Styler.highlight_max: Highlight the maximum with a style.
+        Styler.highlight_quantile: Highlight values defined by a quantile with a style.
+        Styler.highlight_between: Highlight a defined range with a style.
         """
 
         def f(data: FrameOrSeries, props: str) -> np.ndarray:
-            return np.where(data == np.nanmin(data.values), props, "")
+            return np.where(data == np.nanmin(data.to_numpy()), props, "")
+
+        if props is None:
+            props = f"background-color: {color};"
+        return self.apply(f, axis=axis, subset=subset, props=props)
+
+    def highlight_between(
+        self,
+        subset: Optional[IndexLabel] = None,
+        color: str = "yellow",
+        axis: Optional[Axis] = 0,
+        left: Optional[Union[Scalar, Sequence]] = None,
+        right: Optional[Union[Scalar, Sequence]] = None,
+        inclusive: Union[bool, str] = True,
+        props: Optional[str] = None,
+    ) -> Styler:
+        """
+        Highlight a defined range with a style.
+
+        .. versionadded:: 1.3.0
+
+        Parameters
+        ----------
+        subset : IndexSlice, default None
+            A valid slice for ``data`` to limit the style application to.
+        color : str, default 'yellow'
+            Background color to use for highlighting.
+        axis : {0 or 'index', 1 or 'columns', None}, default 0
+            Apply to each column (``axis=0`` or ``'index'``), to each row
+            (``axis=1`` or ``'columns'``), or to the entire DataFrame at once
+            with ``axis=None``.
+        left : scalar or datetime-like, or sequence or array-like, default None
+            Left bound for defining the range.
+        right : scalar or datetime-like, or sequence or array-like, default None
+            Right bound for defining the range.
+        inclusive : str or bool, default True
```

Comment:
look at pd.date_range, inclusie is usualy both, left, right, neither

Reply:
Problem is consistency with Series.between where it is a bool. Perhaps that should be changed first to match date_range in its definition of "inclusive".

Reply:
changed to accept both possibilities. I think Numpy somewhere might use `None` or `'none'` as well. Probably OK for now to have 'some' consistency but not be exactly the same.

`Series.between` does not allow 'left' or 'right', so this might be added as extension. I can add an `good first issue` and see if anyone picks it up